activerecord/lib/active_record/model_schema.rb

```diff
@@ -431,6 +434,34 @@ def attribute_types # :nodoc:
         @attribute_types ||= Hash.new(Type.default_value)
       end
 
+      ##
+      # :method: before_load_schema
+      #
+      # :call-seq: before_load_schema(*args, &block)
+      #
+      # Registers a callback to be called on the record class before the schema is loaded successfully. See
+      # ActiveRecord::ModelSchema for more information.
+      def before_load_schema(*, &block)
+        if block.nil?
+          fail NotImplementedError, "before_load_schema expects a block. It does not supports args"
+        end
+        __before_load_schema_callbacks << [self, block]
+      end
+
+      ##
+      # :method: after_load_schema
+      #
+      # :call-seq: after_load_schema(*args, &block)
+      #
+      # Registers a callback to be called on the record class after the schema is loaded successfully. See
+      # ActiveRecord::ModelSchema for more information.
+      def after_load_schema(*, &block)
+        if block.nil?
+          fail NotImplementedError, "after_load_schema expects a block. It does not supports args"
+        end
+        __after_load_schema_callbacks << [self, block]
```

Comment:
```suggestion
        __after_load_schema_callbacks += [[self, block]]
```

Because the default value is shared, so if you inherit the whole chain share the same callback list.